InnerEye/ML/deep_learning_config.py

```diff
@@ -201,12 +201,29 @@ class WorkflowParams(param.Parameterized):
                                                           "associated with when performing k-fold cross validation")
     perform_training_set_inference: bool = \
         param.Boolean(False,
-                      doc="If True, run full image inference on the training set at the end of training. If False and "
-                          "perform_validation_and_test_set_inference is True (default), only run inference on "
-                          "validation and test set. If both flags are False do not run inference.")
+                      doc="If True, run full image inference on the training set at the end of training.")
     perform_validation_and_test_set_inference: bool = \
+        param.Boolean(None,
+                      doc="If True, run full image inference on validation and test sets after training.")
+    perform_validation_set_inference: bool = \
         param.Boolean(True,
-                      doc="If True (default), run full image inference on validation and test set after training.")
+                      doc="If True (default), run full image inference on validation set after training.")
+    perform_test_set_inference: bool = \
```

Comment:
all these flag names have to be typed on the commandline. Maybe shorten them a bit? `inference_on_train_set`, `...val_set`, `...test_set`, `ensemble_inference_on_train_set`

Reply:
Also, for choosing reasonable defaults, I would make all those flags Optional[bool]. Then it's easy to see if a flag has been set manually, and skip it in the automatic choice of defaults.

Reply:
Shorter names and all now optional